crates/iota-genesis-builder/src/stardust/migration/executor.rs

```diff
@@ -316,7 +318,17 @@ impl Executor {
         let move_alias = iota_types::stardust::output::Alias::try_from_stardust(alias_id, alias)?;
 
         // TODO: We should ensure that no circular ownership exists.
-        let alias_output_owner = stardust_to_iota_address_owner(alias.governor_address())?;
+        let mut alias_output_owner = stardust_to_iota_address_owner(alias.governor_address())?;
```

Comment:
In order to reduce the changes, the function `stardust_to_iota_address_owner_maybe_swap` can be created and called here. This new function takes as input `addresss_swap_map`, invokes `stardust_to_iota_address_owner` and then from the result it executes the code below (`if let Some(owner) = ...`).

Same for `stardust_to_iota_address` -> `stardust_to_iota_address_maybe_swap`.

Reply:
Fixed